EIPS/eip-5806.md

```diff
@@ -64,10 +64,17 @@ For security reasons, some opcodes should not be executed in the context of an E
 
 - `SSTORE` (0x55): Setting storage under an EOA breaks many assumptions. In particular storage set through a delegate transaction could cause issues if the accounts later "migrates" using [EIP-7377](./eip-7377.md) or similar. Additionally, storage may be a source of conflicts if a single EOA uses delegate transactions to target codes that interpret the storage layout under this account differently. For all these reasons, EOA should be forbiden from performing `SSTORE` in the context of a delegate transaction. If a delegate transaction performs a CALL, the target of the call is free to manipulate storage normally.
 
-- `CREATE` (0xF0), `CREATE2` (0xF5) and `SELFDESTRUCT` (0xFF): There may be an expectation that transactions from a given sender should have consecutive nonces. This assumption would be broken if an EOA was able to execute one or multiple operations that alter the sender account's nonce. Consequently, EOA performing a delegate transaction should not be able to use the `CREATE`, `CREATE2` or `SELFDESTRUCT` opcodes. If a delegate transaction performs a CALL, the target of the call is free to create contracts normally.
+- `CREATE` (0xF0) and `SELFDESTRUCT` (0xFF): There may be an expectation that transactions from a given sender should have consecutive nonces. This assumption would be broken if an EOA was able to execute one or multiple operations that alter the sender account's nonce. Consequently, EOA performing a delegate transaction should not be able to use the `CREATE` or `SELFDESTRUCT` opcodes. If a delegate transaction performs a CALL, the target of the call is free to create contracts normally.
 
 Any attempts to make execute one of these restricted operations will instead throw an exception.
 
+### Opcode modification
+
+Additionally, some opcodes behavior should be slightly modified in the context of an EOA:
+
+- `CREATE2` (0xF5): Similarly to `CREATE`, `CREATE2` does increment the nonce of the caller. However, unlike `CREATE` that depends on the nonce to compute the address of the created contract, and need to increment it so that all contracts created have a proper address, `CREATE2` does not depend on that increment. Therefore, if an EOA tries to deploy a contract using `CREATE2` as part of a delegate transaction, the nonce of the EOA should not be incremented. The rest of the `CREATE2` logic remains unmodified.
```

Comment:
```suggestion
- `CREATE2` (0xF5): Similarly to `CREATE`, `CREATE2` does increment the nonce of the caller. However, unlike `CREATE` that depends on the nonce to compute the address of the created contract, and needs to increment it so that all contracts created have a proper address, `CREATE2` does not depend on that increment. Therefore, if an EOA tries to deploy a contract using `CREATE2` as part of a delegate transaction, the nonce of the EOA should not be incremented. The rest of the `CREATE2` logic remains unmodified.
```